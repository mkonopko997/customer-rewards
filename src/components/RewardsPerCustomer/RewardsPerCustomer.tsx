import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  CircularProgress,
} from "@mui/material";
import { useRewardsPerCustomer } from "@/hooks/useRewardsPerCustomer/useRewardsPerCustomer";
import { Link } from "react-router-dom";

export const RewardsPerCustomer = () => {
  const rewards = useRewardsPerCustomer();

  if (!rewards) {
    return (
      <div className="w-full flex justify-center">
        <CircularProgress className="text-center" />
      </div>
    );
  }

  return (
    <div>
      <TableContainer component={Paper}>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Customer</TableCell>
              <TableCell>Reward</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {Object.keys(rewards).map((key) => (
              <TableRow key={key}>
                <TableCell>
                  <Link to={key} className="text-blue-500 hover:underline">
                    {key}
                  </Link>
                </TableCell>
                <TableCell>{rewards[key]}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
};
