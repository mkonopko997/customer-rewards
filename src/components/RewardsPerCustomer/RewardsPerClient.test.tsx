import { render, waitFor } from "@/test/testUtils/testUtils";
import { screen } from "@testing-library/react";
import { RewardsPerCustomer } from "@/components/RewardsPerCustomer/RewardsPerCustomer";
import customers from "@/test/mocks/customers.json";

describe("RewardsPerCustomer", () => {
  it("should display table of customers and rewards", async () => {
    const { container } = render(<RewardsPerCustomer />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("should display links to rewards per month page", async () => {
    render(<RewardsPerCustomer />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    screen.getAllByRole("link").forEach((link, index) => {
      expect(link.getAttribute("href")).toEqual(`/${customers[index]}`);
    });
  });
});
