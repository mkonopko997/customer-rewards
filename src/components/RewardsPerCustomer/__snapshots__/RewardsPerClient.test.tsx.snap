// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsPerCustomer should display table of customers and rewards 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Customer
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Reward
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <a
                class="text-blue-500 hover:underline"
                href="/testCustomer1"
              >
                testCustomer1
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              400
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <a
                class="text-blue-500 hover:underline"
                href="/testCustomer2"
              >
                testCustomer2
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              230
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <a
                class="text-blue-500 hover:underline"
                href="/testCustomer3"
              >
                testCustomer3
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              950
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
